scipy/stats/_stats_py.py

```diff
@@ -7666,11 +7694,12 @@ def _calc_t_stat(a, b, equal_var, axis=-1):
     var_b = _var(b, axis=axis, ddof=1)
 
     if not equal_var:
-        denom = _unequal_var_ttest_denom(var_a, na, var_b, nb)[1]
+        _, denom = _unequal_var_ttest_denom(var_a, na, var_b, nb)
     else:
-        denom = _equal_var_ttest_denom(var_a, na, var_b, nb)[1]
+        _, denom = _equal_var_ttest_denom(var_a, na, var_b, nb)
 
-    return (avg_a-avg_b)/denom
+    res = (avg_a-avg_b)/denom
+    return res
```

Comment:
This can be changed back if desired, but it's not worth re-running CI.
```suggestion
    return (avg_a-avg_b)/denom
```
The reason is that I was going to do `res[()] if res.ndim == 0 else res` but it didn't seem to be needed.